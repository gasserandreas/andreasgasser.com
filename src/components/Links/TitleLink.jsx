import styled from 'styled-components';

import Link, { propTypes, defaultProps } from './Link';

const TitleLink = styled(Link)(({ theme }) => `
  color: ${theme.colors.text};
  font-weight: 900;
  display: inline-block;
  margin-top: 0.25rem;
  margin-bottom: 0.25rem;
  padding-bottom: 0.25rem;

  &:hover {
    color: ${theme.colors.text};
    border-bottom: 2px solid ${theme.colors.text};
  }
`);

TitleLink.propTypes = {
  ...propTypes,
};

TitleLink.defaultProps = {
  ...defaultProps,
};

export default TitleLink;
