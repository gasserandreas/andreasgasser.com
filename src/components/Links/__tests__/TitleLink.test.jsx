/* global withTheme */
import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';

import TitleLink from '../TitleLink';

describe('components/TitleLink', () => {
  const defaultProps = {
    to: 'https://www.google.com/',
    external: true,
    children: 'click me',
  };

  it('simple snapshot test', () => {
    const dom = mount(withTheme(<TitleLink {...defaultProps} />));
    expect(toJson(dom)).toMatchSnapshot();
  });
});
