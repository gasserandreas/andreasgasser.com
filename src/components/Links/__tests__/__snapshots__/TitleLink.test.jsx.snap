// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/TitleLink simple snapshot test 1`] = `
<CustomThemeProvider
  dark={false}
>
  <ThemeProvider
    theme={
      Object {
        "breakpoints": Array [
          "320px",
          "540px",
          "720px",
          "960px",
          "1140px",
          "1400px",
        ],
        "buttons": Object {
          "outline": Object {
            "bg": "transparent",
            "boxShadow": "inset 0 0 2px",
            "color": "primary",
            "variant": "buttons.primary",
          },
          "primary": Object {
            "bg": "primary",
            "borderRadius": "default",
            "color": "background",
            "fontSize": 2,
            "fontWeight": "bold",
          },
          "secondary": Object {
            "bg": "secondary",
            "color": "background",
            "variant": "buttons.primary",
          },
        },
        "colors": Object {
          "background": "#c3c3c3",
          "backgroundInverse": "#fff",
          "gray": "#dddddf",
          "highlight": "hsla(205, 100%, 40%, 0.125)",
          "muted": "#c9c9c9",
          "primary": "#07c",
          "secondary": "#30c",
          "text": "#333",
          "textHighlighed": "#565656",
          "textHover": "#7e7e7e",
          "textInverse": "#eee",
        },
        "fontSizes": Array [
          12,
          14,
          16,
          20,
          24,
          32,
          48,
          64,
          96,
        ],
        "fontWeights": Object {
          "body": 400,
          "bold": 700,
          "heading": 700,
        },
        "fonts": Object {
          "body": "system-ui, sans-serif",
          "heading": "inherit",
          "monospace": "Menlo, monospace",
        },
        "lineHeights": Object {
          "body": 1.5,
          "heading": 1.25,
        },
        "radii": Object {
          "circle": 99999,
          "default": 4,
        },
        "shadows": Object {
          "card": "0 0 4px rgba(0, 0, 0, .125)",
        },
        "sizes": Object {
          "avatar": 48,
        },
        "space": Array [
          0,
          4,
          8,
          16,
          32,
          64,
          128,
          256,
          512,
        ],
        "styles": Object {
          "root": Object {
            "fontFamily": "body",
            "fontWeight": "body",
            "lineHeight": "body",
          },
        },
        "text": Object {
          "caps": Object {
            "letterSpacing": "0.1em",
            "textTransform": "uppercase",
          },
          "display": Object {
            "fontFamily": "heading",
            "fontSize": Array [
              5,
              6,
              7,
            ],
            "fontWeight": "heading",
            "lineHeight": "heading",
          },
          "heading": Object {
            "fontFamily": "heading",
            "fontWeight": "heading",
            "lineHeight": "heading",
          },
        },
        "variants": Object {
          "avatar": Object {
            "borderRadius": "circle",
            "height": "avatar",
            "width": "avatar",
          },
          "card": Object {
            "bg": "background",
            "boxShadow": "card",
            "p": 2,
          },
          "link": Object {
            "color": "primary",
          },
          "nav": Object {
            ":hover,:focus,.active": Object {
              "color": "primary",
            },
            "color": "inherit",
            "display": "inline-block",
            "fontSize": 1,
            "fontWeight": "bold",
            "p": 2,
            "textDecoration": "none",
          },
        },
      }
    }
  >
    <Styled(Link)
      external={true}
      to="https://www.google.com/"
    >
      <StyledComponent
        external={true}
        forwardedComponent={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "_foldedDefaultProps": Object {
              "external": false,
            },
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "sc-htpNat",
              "isStatic": false,
              "lastClassName": "gmsjui",
              "rules": Array [
                [Function],
              ],
            },
            "displayName": "Styled(Link)",
            "foldedComponentIds": Array [],
            "propTypes": Object {
              "children": [Function],
              "external": [Function],
              "to": [Function],
            },
            "render": [Function],
            "styledComponentId": "sc-htpNat",
            "target": [Function],
            "toString": [Function],
            "usesTheme": false,
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          }
        }
        forwardedRef={null}
        to="https://www.google.com/"
      >
        <Link
          className="sc-htpNat gmsjui"
          external={true}
          to="https://www.google.com/"
        >
          <styled.a
            className="sc-htpNat gmsjui"
            href="https://www.google.com/"
            target="_blank"
          >
            <StyledComponent
              className="sc-htpNat gmsjui"
              forwardedComponent={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "attrs": Array [],
                  "componentStyle": ComponentStyle {
                    "componentId": "sc-bwzfXH",
                    "isStatic": false,
                    "lastClassName": "czfiUD",
                    "rules": Array [
                      [Function],
                    ],
                  },
                  "displayName": "styled.a",
                  "foldedComponentIds": Array [],
                  "render": [Function],
                  "styledComponentId": "sc-bwzfXH",
                  "target": "a",
                  "toString": [Function],
                  "usesTheme": false,
                  "warnTooManyClasses": [Function],
                  "withComponent": [Function],
                }
              }
              forwardedRef={null}
              href="https://www.google.com/"
              target="_blank"
            >
              <a
                className="sc-bwzfXH czfiUD sc-htpNat gmsjui"
                href="https://www.google.com/"
                target="_blank"
              >
                click me
              </a>
            </StyledComponent>
          </styled.a>
        </Link>
      </StyledComponent>
    </Styled(Link)>
  </ThemeProvider>
</CustomThemeProvider>
`;
